crates/uv/src/commands/auth/logout.rs

```diff
@@ -0,0 +1,46 @@
+use anyhow::{Context, Result, bail};
+use std::fmt::Write;
+use uv_configuration::{KeyringProviderType, Service};
+
+use crate::{commands::ExitStatus, printer::Printer};
+
+/// Logout from a service.
+///
+/// If no username is provided, defaults to `__token__`.
+pub(crate) async fn logout(
+    service: Service,
+    username: Option<String>,
+    keyring_provider: Option<KeyringProviderType>,
+    printer: Printer,
+) -> Result<ExitStatus> {
+    let url = service.url();
+    let display_url = username
+        .as_ref()
+        .map(|username| format!("{username}@{url}"))
+        .unwrap_or_else(|| url.to_string());
+    let username = username.unwrap_or_else(|| String::from("__token__"));
```

Comment:
If you wanted to get fancy, you could make this `Cow`... Like `username.map(Cow::Owned).unwrap_or(Cow::Borrowed("__token__"))`.

(You def don't need to do this, just thought you mind find it interesting.)